### DISCUSSION TITLE
NodeJS/Angular app issues with Vercel

### DISCUSSION THREAD

**Discussion by Andysimmonsisme** [vercel/community#1300] [Help] [answered]
### Summary

I have deployed my NodeJS app to Vercel (Angular front end resides in public/gorfff and Node index.js file resides in root), and I am getting the following error when the Angular index.html file tries to load the polyfills, runtime, and main js files: Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "text/html". Strict MIME type checking is enforced for module scripts per HTML spec. I've done a lot of Googling and experimented with changing the base href in the index.html file to the absolute path of the three aforementioned js files, but I still get the same error. When I leave the base href="/" locally the website works just fine, however if I change it to the absolute path I get the aforementioned error locally, so that tells me I've got something configured incorrectly that Vercel isn't liking (I'm guessing it has to do with the base href somehow), but I'm all out of options. Any ideas?

Here is the Vercel url: https://gorfff.vercel.app/
Here is my GitHub repo: https://github.com/Andysimmonsisme/gorfff

**Comment by amyegan** [comment, marked as answer]
Since Vercel is serverless, you need to put your Node.js code in a Serverless Function and build the Angular pages for deployment. You'll need to restructure your app a little, or change the build script in your project configuration.

Additional resources:
- [How to Deploy an Angular Site with Vercel](https://vercel.com/guides/deploying-angular-with-vercel)
- [Using Express.js with Vercel](https://vercel.com/guides/using-express-with-vercel)
- [Angular example repo](https://github.com/vercel/vercel/tree/main/examples/angular)
- [Deploying Serverless Functions](https://vercel.com/docs/concepts/functions/serverless-functions#deploying-serverless-functions) (look at the "Node.js" tabs on code example section)
- [Serverless Function Runtimes](https://vercel.com/docs/concepts/functions/serverless-functions/runtimes#)

**Reply by Andysimmonsisme** [reply]
Thanks Amy! I was able to get it working by creating a new application with the root directory as public/gorfff and then creating a serverless function with my NodeJS code (I did have to do a little reworking). Definitely would have taken me a long time to figure this out, so thank you for saving me the time!
